compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/AbstractMultiVersionExtension.java

```diff
@@ -46,10 +49,44 @@
  * <p>Implements extension to handle {@link VersionCondition}.
  */
 abstract class AbstractMultiVersionExtension
-    implements BeforeAllCallback, BeforeEachCallback, ExecutionCondition {
+    implements BeforeAllCallback, BeforeEachCallback, ExecutionCondition, MultiEnvTestExtension {
+
+  private static final String NESSIE_VERSION_SEGMENT_TYPE = "nessie-version";
 
   private static final ConditionEvaluationResult PASS = enabled(null);
 
+  @Override
+  public String segmentType() {
+    return NESSIE_VERSION_SEGMENT_TYPE;
+  }
+
+  @Override
+  public SortedSet<String> allEnvironmentIds(ConfigurationParameters configuration) {
```

Comment:
Not sure whether `SortedSet<String>` is correct here. While the `Version`s are `Comparable`, the set of sorted `Version.toString()` might be different. Maybe use a `Set<String>` as the result type and use a `LinkedHashSet` instead of `TreeSet` as the concrete type?

Reply:
My intention was to ensure a stable iteration over test cases between runs. I'll probably change to `List<String>` with a javadoc comment instead. Implementations can chose their own order.